src/staticanalysis/bot/static_analysis_bot/clang/tidy.py

```diff
@@ -235,6 +235,12 @@ def __init__(self, header_data, revision):
         assert len(header_data) == 6
         assert not settings.repo_dir.endswith('/')
         self.revision = revision
+
+        self.reason = None
+        for c in settings.clang_checkers:
+            if c['name'] == self.check:
```

Comment:
Nit: As you can see below your code, `self.check` is only set to the right value (from `header_data`) *after* this code runs, so it looks like unfortunately your code reads "find all the checkers where `name` is `None`".

To fix this, please move your entire code block below, e.g. at the end of the `__init__` method.